src/test/unit_tests/test_bucketspace_fs.js

```diff
@@ -456,7 +474,33 @@ mocha.describe('bucketspace_fs', function() {
             const res2 = await bucketspace_fs.list_buckets(dummy_object_sdk_for_iam_account2);
             assert.equal(res2.buckets.length, 1);
         });
-        mocha.after(async function() {
+        mocha.it('list buckets - different account', async function() {
+            // another user created a bucket (account_user3 cannot list it)
+            const dummy_object_sdk_for_iam_account = make_dummy_object_sdk_for_account(dummy_object_sdk, account_user3);
+            const res = await bucketspace_fs.list_buckets(dummy_object_sdk_for_iam_account);
+            assert.equal(res.buckets.length, 0);
+        });
+        mocha.it('list buckets - different account with bucket policy (principal by name)', async function() {
+            // another user created a bucket
+            // with bucket policy account_user3 can list it
+            const policy = generate_s3_policy(account_user4.name, test_bucket, ['s3:*']).policy;
+            const param = { name: test_bucket, policy: policy };
+            await bucketspace_fs.put_bucket_policy(param);
+            const dummy_object_sdk_for_iam_account = make_dummy_object_sdk_for_account(dummy_object_sdk, account_user4);
+            const res = await bucketspace_fs.list_buckets(dummy_object_sdk_for_iam_account);
+            assert.equal(res.buckets.length, 1);
+        });
+        mocha.it('list buckets - different account with bucket policy (principal by id)', async function() {
+            // another user created a bucket
+            // with bucket policy account_user3 can list it
+            const policy = generate_s3_policy(account_user4._id, test_bucket, ['s3:*']).policy;
+            const param = { name: test_bucket, policy: policy };
+            await bucketspace_fs.put_bucket_policy(param);
+            const dummy_object_sdk_for_iam_account = make_dummy_object_sdk_for_account(dummy_object_sdk, account_user4);
+            const res = await bucketspace_fs.list_buckets(dummy_object_sdk_for_iam_account);
+            assert.equal(res.buckets.length, 1);
```

Comment:
I think it's better to check the name of the bucket and not just the size of the buckets array received

Reply:
Added, thanks.